### PR TITLE
Summer2023

### PR DESCRIPTION
**Pull request by Dhananjhay**
Following are the list of changes from the previous PR (summer2023 -> master):
1. Added a `stellar_model_dir` argument in the use_gaia_spectrum() function and the Observation() class to do transit simulation; this was done to prevent users from fiddling around with the source code.
2. Reverted back to interpolating the passband to the spectrum resolution in the transit simulation calculation.

### CODE REVIEW

**Comment by BearBearCodes** [comment]
Thank you for this Dhananjhay! I will review this PR properly later this week, if that's alright, as I have an E&M midterm in less than 2 days :'(

In the meantime, would you be able to add a summary of your changes to the CHANGELOG? (No rush, and if necessary it can be done after we merge this PR)

**Comment by Dhananjhay** [comment]
Hey @BearBearCodes! Good luck on your midterm! I'll add a summary of my changes to each repo's CHANGELOG.md once you have reviewed the PR(s). In the meantime, I am going to test the corresponding docker images on the CANFAR server and once that's done, I'll make tutorial on how to use the transit simulation feature using jupyter-notebook interface.

**Comment by BearBearCodes** [comment]
That sounds like a good plan. Thanks again for your incredible work! 😄

File: castor_etc/conversions.py
Comment (BearBearCodes) [inline castor_etc/conversions.py:1]: I'll remove the unnecessary `TelescopeObj` description in the docstring, but other than that it looks good

File: castor_etc/parameters.py
Comment (BearBearCodes) [inline castor_etc/parameters.py:1]: I'll add `# megapixels` to the `TRANSIT_CCD_DIMENSIONS` description after merging

File: docker/apt-install.sh
Comment (BearBearCodes) [inline docker/apt-install.sh:1]: This file should remain unchanged, as it is a general utility that doesn't depend on the specific CASTOR Docker requirements.

To install gcc and g++ in the Docker image, insert the following line
```Dockerfile
RUN ./apt-install.sh gcc g++
```
in [Dockerfile.noCustomEnv](https://github.com/CASTOR-telescope/ETC/blob/8c506215c2664df2a42e4c4389aec7cd64c73ea2/docker/Dockerfile.noCustomEnv#L22) and [Dockerfile.yesCustomEnv](https://github.com/CASTOR-telescope/ETC/blob/8c506215c2664df2a42e4c4389aec7cd64c73ea2/docker/Dockerfile.yesCustomEnv#L29). I can do this after merging

File: README.md
Comment (BearBearCodes) [inline README.md:1]: After merging, I'll add a bit more to the instructions regarding the stellar models directory. Namely, this is only needed if you are doing transit simulations, and the variable to set in spectrum.py is `stellar_model_dir` in `getStarData()` and `use_gaia_spectrum()`.

**Review by BearBearCodes — APPROVED** [review]
Thanks for your patience, @Dhananjhay. Excellent work! I'll merge this PR now. Once you have some time (no rush), could you also write a summary of your changes in the CHANGELOG.md?

**Pull request merged**